src/transformers/pipelines/image_text_to_text.py

```diff
@@ -75,16 +75,32 @@ def retrieve_images_in_messages(
     retrieved_images = []
     for message in messages:
         for content in message["content"]:
-            if isinstance(content, dict) and content.get("type") == "image":
-                if "image" in content:
-                    retrieved_images.append(content["image"])
-                elif idx_images < len(images):
-                    retrieved_images.append(images[idx_images])
-                    idx_images += 1
-                else:
-                    raise ValueError(
-                        "The number of images in the chat messages should be the same as the number of images passed to the pipeline."
-                    )
+            if isinstance(content, dict):
+                if content.get("type") == "image":
+                    if any(key in content for key in ["image", "url", "path"]):
```

Comment:
this can be more efficient, a single for loop should be enough!

Reply:
Just learned about the for...else python structure, makes it cleaner indeed, thanks!